File: ProyectoFinal/Trimestre 5/API REST/src/models/devolucion.model.js
import { Model, DataTypes } from "sequelize";
import sequelize from "../config/connect.db.js";
import Producto from "./producto.model.js";
import Persona from "./persona.model.js";
import TipoDevolucion from "./tipoDevolucion.model.js";

class Devolucion extends Model {}

Devolucion.init(
  {
    idDevolucion: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
    },
    Fecha: {
      type: DataTypes.DATE,
      allowNull: false,
    },
    Motivo: {
      type: DataTypes.STRING(200),
      allowNull: false,
    },
    Cantidad: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    Producto_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: Producto,
        key: "idProducto",
      },
    },
    Persona_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: Persona,
        key: "idPersona",
      },
    },
    TipoDevolucion_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: TipoDevolucion,
        key: "idTipoDevolucion",
      },
    },
  },
  {
    sequelize,
    modelName: "Devolucion",
    tableName: "Devolucion",
    timestamps: false,
  }
);

export default Devolucion;

File: Trimestre 6/Proyecto final/Proyecto final/MOVIL/appmovil/lib/models/devolucion_model.dart
class DevolucionModel {
  final int idDevolucion;
  final String fecha; // yyyy-mm-dd
  final String motivo;
  final int cantidad;
  final int productoId;
  final int tipoDevolucionId;
  final int personaId;

  DevolucionModel({
    required this.idDevolucion,
    required this.fecha,
    required this.motivo,
    required this.cantidad,
    required this.productoId,
    required this.tipoDevolucionId,
    required this.personaId,
  });

  factory DevolucionModel.fromJson(Map<String, dynamic> json) {
    return DevolucionModel(
      idDevolucion: (json['idDevolucion'] ?? 0) as int,
      fecha: (json['Fecha'] ?? '').toString(),
      motivo: (json['Motivo'] ?? '').toString(),
      cantidad: (json['Cantidad'] ?? 0) as int,
      productoId: (json['Producto_id'] ?? 0) as int,
      tipoDevolucionId: (json['TipoDevolucion_id'] ?? 0) as int,
      personaId: (json['Persona_id'] ?? 0) as int,
    );
  }

  Map<String, dynamic> toJsonCreate() {
    return {
      'Fecha': fecha,
      'Motivo': motivo,
      'Cantidad': cantidad,
      'Producto_id': productoId,
      'Persona_id': personaId,
      'TipoDevolucion_id': tipoDevolucionId,
    };
  }

  Map<String, dynamic> toJsonUpdate() {
    return {
      'Fecha': fecha,
      'Motivo': motivo,
      'Cantidad': cantidad,
    };
  }
}
